torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1345,6 +1376,38 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             ++af.pc;
             break;
           }
+          case TYPECHECK: {
+            int num_inputs = stack.back().toInt(), i = 0;
+            pop(stack);
+            TORCH_INTERNAL_ASSERT(stack.size() >= num_inputs);
+            if (num_inputs == 0) {
+              push(stack, true);
+              break;
+            }
+            // Check every input's shape against profiled (expected) shape.
+            for (i = 0; i < num_inputs; i++) {
+              auto& input = peek(stack, i, num_inputs);
+              TORCH_INTERNAL_ASSERT(input.isTensor());
+              auto t = input.toTensor();
+              const TypePtr& expected = af.types[inst.X + i];
+              auto expected_type = expected->cast<TensorType>();
+              if (t.defined() &&
+                  !frames.back().symbols2dims.bindSymbolicShapes(
+                      t.sizes(), expected_type->symbolic_sizes()) &&
+                  !expected_type->matchTensor(t)) {
+                GRAPH_DEBUG(
```

Comment:
This code will be on a hot path, so maybe we should not be using debug-logging here.

Reply:
It is useful to have.   I guess it is ok to remove

Reply:
Yeah, it is useful to have, I agree, but that code will be run every time we execute the op, so we should really try to keep it lean. That's the reason you could see some debug prints are commented out earlier in this function (rather than implemented with GRAPH_DEBUG).